File: frontend/lib/features/challenges/data/models/requests/challenge.dart
class ChallengeUpdateRequestModel {
  final Map<String, String> name;
  final Map<String, String> description;
  final DateTime? startDate;
  final String icon;

  const ChallengeUpdateRequestModel({
    required this.name,
    required this.description,
    required this.startDate,
    required this.icon,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'start_date': startDate?.toUtc().toIso8601String(),
      'icon': icon,
    };
  }
}

class ChallengeCreateRequestModel {
  final Map<String, String> name;
  final Map<String, String> description;
  final DateTime? startDate;
  final String icon;

  const ChallengeCreateRequestModel({
    required this.name,
    required this.description,
    required this.startDate,
    required this.icon,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'start_date': startDate?.toUtc().toIso8601String(),
      'icon': icon,
    };
  }
}

File: backend/api/src/features/challenges/structs/requests/challenge.rs
use std::collections::HashMap;

use chrono::{DateTime, Utc};
use serde::Deserialize;
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetChallengeParams {
    pub challenge_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdateChallengeParams {
    pub challenge_id: Uuid,
}

#[derive(Deserialize)]
pub struct ChallengeDuplicateParams {
    pub challenge_id: Uuid,
}

#[derive(Deserialize)]
pub struct ChallengeUpdateRequest {
    pub name: HashMap<String, String>,
    pub description: HashMap<String, String>,
    pub start_date: Option<DateTime<Utc>>,
    pub icon: String,
}

#[derive(Deserialize)]
pub struct ChallengeCreateRequest {
    pub name: HashMap<String, String>,
    pub description: HashMap<String, String>,
    #[serde(default)]
    pub start_date: Option<DateTime<Utc>>,
    pub icon: String,
}
